carrier/carrier.go

```diff
@@ -95,7 +96,12 @@ func (evc *ethereumCarrier) SubscribeNewBlock(cb func(uint64), close chan bool)
 				close <- closed
 				break
 			case err := <-sub.Err():
-				zap.L().Fatal("error when listening to ethereum block", zap.Error(err))
+				if err := evc.reset(err); err.Error() != "tls: use of closed connection" {
```

Comment:
not sure we should fatal here or not. It seems there is a high possibility we cannot link to server.